jpa/eclipselink.jpa.test/src/org/eclipse/persistence/testing/tests/jpa/beanvalidation/BeanValidationJunitTest.java

```diff
@@ -355,7 +366,34 @@ public void testTraversableResolverPreventsTraversingRelationshipMultipleTimes()
         // If it ever finds an entity with such flag set, the entity has visited the validator twice. It should be flagged as error.
     }
 
-
+    //Bug #411013
+    public void testValidateChangedData() {
+        try {
+            getDatabaseSession().executeNonSelectingSQL("insert into CMP3_BV_PROJECT values (895, \"some long name\")");
+        } catch (Throwable t) {
+            //ignore
+        }
+        clearCache();
+        Map<String, Object> props = new HashMap<>();
+        props.put("eclipselink.weaving", "false");
+        EntityManagerFactory factory = getEntityManagerFactory(props);
+        EntityManager em = factory.createEntityManager();
+        try {
+            beginTransaction(em);
+            TypedQuery<Project> query = em.createQuery("select p from CMP3_BV_PROJECT p", Project.class);
+            for (Project p: query.getResultList()) {
+                System.out.println(p.getName());
+            }
+            commitTransaction(em);
```

Comment:
I'm not sure I understand the test. Validation is happening on transaction commit. Does the validation throw an exception?

Reply:
Yes, exception is thrown but should not be as no entity has been changed.